File: mobile/lib/features/user/infrastructure/models/user_profile_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import 'package:sports_gurukul/features/user/infrastructure/models/address_dto.dart';
import 'package:sports_gurukul/features/user/infrastructure/models/contact_dto.dart';
import 'package:sports_gurukul/features/user/infrastructure/models/user_preference_dto.dart';

part 'user_profile_dto.freezed.dart';
part 'user_profile_dto.g.dart';

/// DTO matching the backend `UserProfileDto` schema.
@freezed
abstract class UserProfileDto with _$UserProfileDto {
  const factory UserProfileDto({
    required String id,
    required String userId,
    required String fullName,
    required String email,
    required String createdAt,
    String? phoneNumber,
    String? dateOfBirth,
    @JsonKey(unknownEnumValue: GenderDto.preferNotToSay)
    @Default(GenderDto.preferNotToSay)
    GenderDto gender,
    String? bio,
    String? profileImageUrl,
    String? coverImageUrl,
    String? height,
    String? weight,
    String? preferredSport,
    String? experienceLevel,
    @JsonKey(unknownEnumValue: UserStatusDto.active)
    @Default(UserStatusDto.active)
    UserStatusDto status,
    @Default(false) bool isEmailVerified,
    String? updatedAt,
    @Default(0) int profileCompletionPercentage,
    @Default([]) List<AddressDto> addresses,
    ContactDto? contactInformation,
    UserPreferenceDto? preferences,
    @Default([]) List<String> roles,
    @Default(true) bool hasProfile,
  }) = _UserProfileDto;

  factory UserProfileDto.fromJson(Map<String, dynamic> json) =>
      _$UserProfileDtoFromJson(json);
}

/// Gender enum matching the backend `Gender`.
@JsonEnum(valueField: 'value')
enum GenderDto {
  @JsonValue(0)
  male,
  @JsonValue(1)
  female,
  @JsonValue(2)
  nonBinary,
  @JsonValue(3)
  preferNotToSay,
}

/// User status enum matching the backend `UserStatus`.
@JsonEnum(valueField: 'value')
enum UserStatusDto {
  @JsonValue(0)
  active,
  @JsonValue(1)
  inactive,
  @JsonValue(2)
  suspended,
  @JsonValue(3)
  locked,
}

File: backend/src/SportsGurukul.Application/Features/UserManagement/DTOs/UserProfileDto.cs
using SportsGurukul.Domain.Enums;

namespace SportsGurukul.Application.Features.UserManagement.DTOs;

public class UserProfileDto
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public string FullName { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    public string? PhoneNumber { get; set; }
    public DateTime? DateOfBirth { get; set; }
    public Gender Gender { get; set; }
    public string? Bio { get; set; }
    public string? ProfileImageUrl { get; set; }
    public string? CoverImageUrl { get; set; }
    public string? Height { get; set; }
    public string? Weight { get; set; }
    public string? PreferredSport { get; set; }
    public string? ExperienceLevel { get; set; }
    public UserStatus Status { get; set; }
    public bool IsEmailVerified { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public int ProfileCompletionPercentage { get; set; }
    public IReadOnlyList<AddressDto> Addresses { get; set; } = [];
    public ContactDto? ContactInformation { get; set; }
    public UserPreferenceDto? Preferences { get; set; }
    public IReadOnlyList<string> Roles { get; set; } = [];

    /// <summary>
    /// Whether the user has a completed <c>UserProfile</c> yet. False when the
    /// endpoint answers with an identity-only payload because no profile
    /// exists; roles are always current regardless of this flag.
    /// </summary>
    public bool HasProfile { get; set; } = true;
}
